service/resource/legacy/resource.go

```diff
@@ -173,57 +177,34 @@ type Event struct {
 	Object *awstpr.CustomObject
 }
 
-func (s *Service) Boot() {
-	s.bootOnce.Do(func() {
-		err := s.tpr.CreateAndWait()
-		if tpr.IsAlreadyExists(err) {
-			s.logger.Log("debug", "third party resource already exists")
-		} else if err != nil {
-			s.logger.Log("error", fmt.Sprintf("%#v", err))
-			return
-		}
-
-		s.logger.Log("debug", "starting list/watch")
-
-		newResourceEventHandler := &cache.ResourceEventHandlerFuncs{
-			AddFunc:    s.addFunc,
-			DeleteFunc: s.deleteFunc,
-			UpdateFunc: s.updateFunc,
-		}
-		newZeroObjectFactory := &tpr.ZeroObjectFactoryFuncs{
-			NewObjectFunc:     func() runtime.Object { return &awstpr.CustomObject{} },
-			NewObjectListFunc: func() runtime.Object { return &awstpr.List{} },
-		}
-
-		s.tpr.NewInformer(newResourceEventHandler, newZeroObjectFactory).Run(nil)
-	})
-}
-
-func (s *Service) addFunc(obj interface{}) {
+// NewUpdatePatch is called upon observed custom object change. It creates the
+// AWS resources for the cluster.
+func (s *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
 	customObject, ok := obj.(*awstpr.CustomObject)
 	if !ok {
-		s.logger.Log("error", "could not convert to awstpr.CustomObject")
-		return
+		return &framework.Patch{}, microerror.Maskf(invalidConfigError, "could not convert to awstpr.CustomObject")
```

Comment:
Zero value of a pointer is `nil`. Makes the return value easier.

Reply:
Good point. I'm returning nil in all except the final return.